File: bob-client/src/main/scala/fr/dmconcept/bob/client/models/BoardConfig.scala
package fr.dmconcept.bob.client.models

import fr.dmconcept.bob.client.models.json.BobJsonProtocol._
import java.util.UUID
import spray.json._

object BoardConfig extends {

  def apply(name: String, servoConfigs: Vector[ServoConfig]): BoardConfig = {
    BoardConfig(UUID.randomUUID.toString, name, servoConfigs)
  }

  def deserialize(json: JsValue) = json.convertTo[BoardConfig]

}

case class BoardConfig(

    id           : String,

    name         : String,

    servoConfigs : Vector[ServoConfig]

) {

  assert(!name.isEmpty        , "Empty name"         )
  assert(servoConfigs.nonEmpty, "Empty servo configs")

}

File: bob-server_gradle/app/src/main/java/fr/dmconcept/bob/server/models/BoardConfig.java
package fr.dmconcept.bob.server.models;

import java.util.List;

public class BoardConfig {

    public String id;

    public String name;

    public List<ServoConfig> servoConfigs;

    public BoardConfig(String id, String name, List<ServoConfig> servoConfigs) {
        this.id           = id;
        this.name         = name;
        this.servoConfigs = servoConfigs;
    }

}
